library/std/src/env.rs

```diff
@@ -872,19 +874,28 @@ impl !Sync for Args {}
 #[stable(feature = "env", since = "1.0.0")]
 impl Iterator for Args {
     type Item = String;
+
     fn next(&mut self) -> Option<String> {
         self.inner.next().map(|s| s.into_string().unwrap())
     }
+
+    #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
         self.inner.size_hint()
     }
+
+    // Methods which skip args cannot simply delegate to the inner iterator,
+    // because that would bypass UTF-8 validation.
```

Comment:
I am... pretty sure these two lines are explaining what the PR description seems to swing-and-whiff at, but I would still like to have it clearly noted in the PR description so I can better understand why I... yeah.

Reply:
I've reworded the description.